lib/client.js

```diff
@@ -216,6 +224,34 @@ class ServiceClient {
             timeout: DEFAULT_REQUEST_TIMEOUT
         }, this.options.defaultRequestOptions);
 
+        this.options.retryOptions = Object.assign({
+            retries: 0,
+            factor: 2,
+            minTimeout: 200,
+            maxTimeout: 400,
+            randomize: true,
+            // By default all errors are indicated as being transient
+            // users should define their own check in order to handle errors as they require
+            // it takes the error as an argument
+            transientErrorCheck(err) {
+                // by default don't retry if the circuit breaker is open
+                if (err.type === ServiceClient.CIRCUIT_OPEN) {
+                    return false;
+                }
+                return true;
+            },
+            // this is called each time a retry is triggered
+            // it receives the current retry attempt and error as args
+            // eslint-disable-next-line
+            onRetry(currentAttempt, err) {
+                return currentAttempt;
```

Comment:
Why do you return this?

Reply:
It is just a default placeholder, do you think an empty function would be better?

Reply:
Yes, if we return something here, people will think that they also should return something.

Reply:
Agreed